azurelinuxagent/common/utils/shellutil.py

```diff
@@ -110,6 +113,37 @@ def run_get_output(cmd, chk_err=True, log_cmd=True, expected_errors=[]):
     return 0, output
 
 
+def _encode_command_output(output):
+    return ustr(output, encoding='utf-8', errors="backslashreplace")
+
+
+def run_command(command):
+    """
+    Wrapper for subprocess.Popen. Executes the given command and returns its stdout.
+    The command parameter is a list of strings, e.g. ["ps" "aux"].
+    Logs any errors executing the command and raises an exception.
+    """
+    retcode = 0
+
+    try:
+        process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=False)
+        stdout, stderr = process.communicate()
+        retcode = process.returncode
+    except Exception as e:
+        logger.error(u"Command [{0}] raised unexpected exception: [{1}]".format(command, ustr(e)))
```

Comment:
It is not the command who raised the exception (the command was not executed). Maybe something like "Cannot execute {0}. Error: {1}"?

Reply:
Updated.